perf/lock.ts

```diff
@@ -0,0 +1,93 @@
+import {assert} from './replicache.js';
+import {newTab} from '../src/test-util';
+import {uuid} from '../src/sync/uuid.js';
+
+enum Mode {
+  SameTab = '',
+  OtherTab = 'other tab',
+  OtherProcess = 'other process',
+}
+
+export function benchmarks() {
+  return [
+    benchmarkUncontendedRead(),
+    benchmarkHeldRead(Mode.SameTab),
+    benchmarkHeldRead(Mode.OtherTab),
+    benchmarkHeldRead(Mode.OtherProcess),
+    benchmarkUncontendedWrite(),
+  ];
+}
+
+export async function isHeld(name: string) {
+  return await navigator.locks.request(
+    name,
+    {ifAvailable: true},
+    async lock => lock === null,
+  );
+}
+
+export function lockShared(name: string) {
+  const p = new Promise(res => {
+    window['resolve'] = res;
+  });
+  navigator.locks.request(name, {mode: 'shared'}, async () => p);
```

Comment:
useless `async`

Reply:
Semantically, yes, but Typescript insists. :)

Reply:
From the type system's view there is no difference between the following two:

```ts
const f1 = async () => 42;
const f2 = () => Promise.resolve(42);
```

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZgRhgXhgQwgTzMGAKAShQD4YAWAJgG4AoUSWOClfI5UgBQCcQBbASwgBTAHRchEEABsAbkLyUCtOuGjwAzAC4YUTAAchIOPCSpEtemrhltug0ZMsmF1YwCst-YeNMnCFwzwAGye9j7MZtRAA

Reply:
Something else must be the issue.